test/com/cloudogu/ces/cesbuildlib/ScriptMock.groovy

```diff
@@ -28,21 +31,61 @@ class ScriptMock {
     List<String> actualWithEnv
     String actualDir
     def actualGitArgs
+    private ignoreOutputFile
+
+    public ScriptMock() {
+        this(true)
+    }
+
+    public ScriptMock(boolean ignoreOutputFile) {
+        // This flag was added to make testing more convenient because we don't need to check for this in most tests.
+        // When set to true, the script command "cat output" and "rm -f output" will not be saved in output
+        // Also the " > output" at the end of each git command will be removed.
+        this.ignoreOutputFile = ignoreOutputFile
+    }
 
     String sh(String args) {
-        actualShStringArgs.add(args.toString())
+        if (this.shouldAddToArgs(args)) {
+            actualShStringArgs.add(this.convert(args))
+            allActualArgs.add(this.convert(args))
+        }
         return getReturnValueFor(args)
     }
 
     String sh(Map<String, Object> args) {
-        actualShMapArgs.add(args.script.toString())
-        return getReturnValueFor(args.get('script'))
+        def script = args.get('script')
+
+        if (this.shouldAddToArgs(script)) {
+            actualShMapArgs.add(this.convert(script))
+            allActualArgs.add(this.convert(script))
+        }
+
+        return getReturnValueFor(script)
+    }
+
+    private String convert(args) {
+        def converted = args.toString()
+
+        if (!this.ignoreOutputFile || converted.indexOf(" > output") == -1) {
+            return converted
+        }
+
+        return converted.substring(0, converted.indexOf(" > output"))
+    }
```

Comment:
I am not sure if this is a good practice. But i encountered that with the new implementation of the executeGit method testing was very very inconvenient because there was a " > output" after every git command and i had to update all tests.
Also there were a "cat output" and "rm -f output" after every git excution. 
That is why i updated the mock to (optionally) remove these things to make testing more convenient.
But is this a good practice or should i remove this again?

Reply:
We can avoid this as described in my comment for `Git.grooy` bellow.